src/coreclr/jit/lowerxarch.cpp

```diff
@@ -7387,6 +7523,87 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
             return false;
         }
 
+        case NI_SSE3_MoveAndDuplicate:
+        case NI_AVX2_BroadcastScalarToVector128:
+        case NI_AVX2_BroadcastScalarToVector256:
+        case NI_AVX512F_BroadcastScalarToVector512:
+        {
+            if (intrinsicId == NI_SSE3_MoveAndDuplicate)
+            {
+                // NI_SSE3_MoveAndDuplicate is for Vector128<double> only.
+                assert(childNode->AsHWIntrinsic()->GetSimdBaseType() == TYP_DOUBLE);
+            }
+            if (comp->compOpportunisticallyDependsOn(InstructionSet_AVX512F_VL) &&
+                parentNode->OperIsEmbBroadcastCompatible())
+            {
+
+                GenTree* CreateScalar = childNode->AsHWIntrinsic()->Op(1);
+                if (CreateScalar->OperIs(GT_HWINTRINSIC) &&
+                    CreateScalar->AsHWIntrinsic()->GetHWIntrinsicId() == NI_Vector128_CreateScalarUnsafe)
+                {
+                    GenTree* Scalar = CreateScalar->AsHWIntrinsic()->Op(1);
+                    if (Scalar->OperIs(GT_LCL_VAR))
+                    {
+                        switch (Scalar->TypeGet())
+                        {
+                            case TYP_FLOAT:
+                            case TYP_DOUBLE:
+                            {
+                                const unsigned opLclNum = Scalar->AsLclVar()->GetLclNum();
+                                comp->lvaSetVarDoNotEnregister(
+                                    opLclNum DEBUGARG(DoNotEnregisterReason::LiveInOutOfHandler));
+                                MakeSrcContained(CreateScalar, Scalar);
+                                MakeSrcContained(childNode, CreateScalar);
+                                return true;
+                            }
+
+                            default:
+                                return false;
+                                ;
+                        }
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                }
+                else if (CreateScalar->OperIs(GT_LCL_VAR))
+                {
+                    // if the operand of the CreateScalarUnsafe node is in Integer type, CreateScalarUnsafe node will be
+                    // fold, we need to specially handle this case.
+                    assert(CreateScalar->TypeIs(TYP_INT) || CreateScalar->TypeIs(TYP_UINT) ||
+                           CreateScalar->TypeIs(TYP_LONG) || CreateScalar->TypeIs(TYP_ULONG));
+                    const unsigned opLclNum = CreateScalar->AsLclVar()->GetLclNum();
+                    comp->lvaSetVarDoNotEnregister(opLclNum DEBUGARG(DoNotEnregisterReason::LiveInOutOfHandler));
+                    MakeSrcContained(childNode, CreateScalar);
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
```

Comment:
Maybe you can remove all the `else` cases and just let them fall through to a single `return false;` at the end?